content/tsColinPrecip.Rmd

```diff
@@ -43,37 +46,31 @@ Function to retrieve precip data using [`geoknife`](https://github.com/USGS-R/ge
 
 ```{r getPrecip, message=FALSE}
 getPrecip <- function(states, startDate, endDate){
-  
-  wg_s <- webgeom(geom = 'derivative:US_Counties', attribute = 'STATE')
-  wg_c <- webgeom(geom = 'derivative:US_Counties', attribute = 'COUNTY')
-  wg_f <- webgeom(geom = 'derivative:US_Counties', attribute = 'FIPS')
-  county_info <- data.frame(state = query(wg_s, 'values'), county = query(wg_c, 'values'), 
-                            fips = query(wg_f, 'values'), stringsAsFactors = FALSE) %>% 
-    unique() 
-  
-  counties_fips <- county_info %>% filter(state %in% states) %>%
-    mutate(state_fullname = tolower(state.name[match(state, state.abb)])) %>%
-    mutate(county_mapname = paste(state_fullname, tolower(county), sep=",")) %>%
-    mutate(county_mapname = unlist(strsplit(county_mapname, split = " county")))
+ 
+  # use fips data from maps package
+  counties_fips <- maps::county.fips %>% 
+    mutate(statecounty=as.character(polyname)) %>% # character to split into state & county
+    tidyr::separate(polyname, c('statename', 'county'), ',') %>%
+    mutate(fips = sprintf('%05d', fips)) %>% # fips need 5 digits to join w/ geoknife result
+    filter(statename %in% states) 
   
   stencil <- webgeom(geom = 'derivative:US_Counties',
                      attribute = 'FIPS',
                      values = counties_fips$fips)
```

Comment:
doesn't matter much because the result returned is unique fips, but the length of `counties_fips$fips` is 442, but with 439 unique codes. You could do a unique on this for completeness